packages/fxa-settings/src/pages/ResetPassword/ConfirmTotpResetPassword/index.tsx

```diff
@@ -0,0 +1,152 @@
+  FormAttributes,
+  InputModeEnum,
+} from '../../../components/FormVerifyCode';
+
+export type ConfirmTotpResetPasswordProps = {
+  verifyCode: (code: string) => Promise<void>;
+  verifyRecoveryCode: (code: string) => Promise<void>;
+  codeErrorMessage: string;
+  setCodeErrorMessage: Dispatch<SetStateAction<string>>;
+};
+
+const ConfirmTotpResetPassword = ({
+  verifyCode,
+  verifyRecoveryCode,
+  codeErrorMessage,
+  setCodeErrorMessage,
+}: ConfirmTotpResetPasswordProps) => {
+  const ftlMsgResolver = useFtlMsgResolver();
+  const [showRecoveryCode, setShowRecoveryCode] = useState<boolean>(false);
+
+  const totpFormAttributes: FormAttributes = {
+    inputFtlId: 'confirm-totp-reset-password-input-label',
+    inputLabelText: 'Enter code',
+    pattern: '[0-9]{6}',
+    maxLength: 6,
+    submitButtonFtlId: 'confirm-totp-reset-password-confirm-button',
+    submitButtonText: 'Confirm',
+  };
+
+  const recoveryCodeFormAttributes: FormAttributes = {
+    inputFtlId: 'confirm-recovery-code-reset-password-input-label',
+    inputLabelText: 'Enter 10-digit backup authentication code',
+    inputMode: InputModeEnum.text,
+    pattern: '[a-zA-Z0-9]',
+    maxLength: 10,
+    submitButtonFtlId: 'confirm-totp-reset-password-confirm-button',
+    submitButtonText: 'Confirm',
+  };
+
+  const localizedCustomCodeRequiredMessage = ftlMsgResolver.getMsg(
+    'signin-totp-code-required-error',
+    'Authentication code required'
+  );
+
+  return (
+    <div>
+      {showRecoveryCode ? (
+        <AppLayout>
+          <h1 className="text-grey-400 mb-6 text-left">
+            <FtlMsg id="confirm-totp-reset-password-header">
+              Reset your password
+            </FtlMsg>
+          </h1>
+
+          <h2 className="font-bold text-xl text-left">
+            <FtlMsg id="confirm-totp-reset-password-subheader">
+              Enter your backup recovery code
+            </FtlMsg>
+          </h2>
+
+          <div className="flex space-x-4">
+            <img src={protectionShieldIcon} alt="" />
+            <FtlMsg id="confirm-totp-reset-password-instruction">
+              <p id="totp-code-instruction" className="my-5 text-md text-left">
+                Check your download or saved backup recovery code.
+              </p>
+            </FtlMsg>
+          </div>
+
+          <FormVerifyCode
+            {...{
+              formAttributes: recoveryCodeFormAttributes,
+              verifyCode: verifyRecoveryCode,
+              localizedCustomCodeRequiredMessage,
+              viewName: 'confirm-recovery-code-reset-password',
+              codeErrorMessage,
+              setCodeErrorMessage,
+            }}
+          />
+          <div className="mt-5 link-blue text-sm flex justify-end">
+            <FtlMsg id="confirm-recovery-code-reset-password-trouble-code">
+              <div
+                onClick={() => {
+                  setShowRecoveryCode(false);
+                }}
+              >
+                Back
```

Comment:
I can't seem to view the Figma link right now, but just noting that this looks significantly different than the 2FA code from the login screen:

<img width="612" alt="image" src="https://github.com/user-attachments/assets/ac90b513-1e8f-4ab3-a860-d9cc60cfb5c5">

Reply:
The design for this page was significantly modified because of confusion between the many codes entered during the password reset flow (the pages looked too visually similar). We'll probably want to standardize for 2FA during password reset and sign in, but maybe that should be a follow up (or handled during upcoming work for SMS).